JumpscaleCore/servers/threebot/tests/test_package_manager.py

```diff
@@ -0,0 +1,169 @@
+from unittest import TestCase, skip
+from uuid import uuid4
+
+from Jumpscale import j
+from parameterized import parameterized
+from loguru import logger
+
+LOGGER = logger
+LOGGER.add("PACKAGE_MANAGER_{time}.log")
+
+PACKAGE_NAME = "test_package"
+
+
+class TestPackageManager(TestCase):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.result_path = j.sal.fs.joinPaths(j.sal.fs.getDirName(__file__), PACKAGE_NAME, "result")
+        self.gedis_client = j.servers.threebot.local_start_default(timeout=1200)
+        self.package_manager = self.gedis_client.actors.package_manager
+
+    @staticmethod
+    def random_string():
+        return str(uuid4())[:10]
+
+    @staticmethod
+    def info(message):
+        LOGGER.info(message)
+
+    def get_package(self, package_name):
+        packages = self.package_manager.packages_list().packages
+        for package in packages:
+            if package.name == package_name:
+                return package
+        return None
+
+    def tearDown(self):
+        self.package_manager.package_delete(PACKAGE_NAME)
+        j.sal.fs.remove(self.result_path)
+        super().tearDown()
+
+    @classmethod
+    def tearDownClass(cls):
+        j.servers.threebot.default.stop()
+        super().tearDownClass()
+
+    @parameterized.expand(["path", "giturl"])
+    def test_001_package_add(self, method):
+        """
+        Test case for adding package (test package) in threebot server
+
+        **Test scenario**
+        #. Start threebot server.
+        #. Add test package.
+        #. Check that the test package has been added.
+        """
+        self.info("Add test package.")
+        if method == "path":
+            path = j.sal.fs.joinPaths(j.sal.fs.getDirName(__file__), PACKAGE_NAME)
+            package = {"path": path}
+        else:
+            giturl = "https://github.com/threefoldtech/jumpscaleX_core/tree/development/JumpscaleCore/serbers/threebot/tests/test_package"
+            package = {"git_url": giturl}
+
+        self.package_manager.package_add(**package)
+        self.gedis_client.reload()
+
+        self.info("Check that the test package has been added.")
+        content = j.sal.fs.readFile(self.result_path)
+        self.assertIn("preparing package", content)
+        self.assertIn("starting packag", content)
+
+    @skip("https://github.com/threefoldtech/jumpscaleX_core/issues/183")
+    def test_002_package_list_delete(self):
+        """
+        Test case for listing and deleting packages
+        
+        **Test scenario**
+        #. Start threebot server.
+        #. Add test package.
+        #. List packages, test package should be found.
+        #. Delete test package.
+        #. List packages again, test package should not be found.
+        """
+        self.info("Add test package.")
+        path = j.sal.fs.joinPaths(j.sal.fs.getDirName(__file__), PACKAGE_NAME)
```

Comment:
this `path` is repeated in all tests, why not just set it in the `__init__`

Reply:
Done here 858aacc8dfa1a058d884ff73f6e043979509325a